### PR TITLE
[FLINK-29105][k8s] Fix the unstable k8s test 'testAddAndLockShouldNotThrowAlreadyExistExceptionWithSameContents'

### PR DESCRIPTION
**Pull request by wangyang0918**
## What is the purpose of the change

Fix the unstable k8s test `testAddAndLockShouldNotThrowAlreadyExistExceptionWithSameContents`

## Verifying this change

* Already covered by the existing UT

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesStateHandleStoreTest.java [hunk 1]
```diff
@@ -1208,14 +1208,16 @@ private static CompletableFuture<Boolean> retryWithFirstFailedK8sOperation(
...
+                                        },
+                                        Executors.newDirectExecutorService()),
                         KubernetesConfigOptions.KUBERNETES_TRANSACTIONAL_OPERATION_MAX_RETRIES
                                 .defaultValue(),
                         t ->
                                 ExceptionUtils.findThrowable(t, KubernetesClientException.class)
                                         .isPresent(),
                         Executors.newDirectExecutorService());
         assertThat(callbackInvocationCount.get()).isEqualTo(2);
+        assertThat(result.isDone()).isTrue();
```
Comment (XComp) [inline flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesStateHandleStoreTest.java:1220]:
```suggestion
        assertThat(result).isDone();
```
nit: just as a fyi - assertj has also asserts for the `CompletableFuture` state. I really have no strong opinion on what looks better, though. ¯\_(ツ)_/¯

Comment (XComp) [reply flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesStateHandleStoreTest.java:1220]:
```suggestion
        assertThat(result).isCompleted();
```
But that said, we can go for a stricter assert here. Check for completion is what we want. `isDone()` would also cover completed exceptionally cases.

**Review by XComp — APPROVED** [review]
LGMT :+1: Interestingly, I ran the test before the fix repeatedly (10000 iterations or something as far as I remember) in the IDE without the race condition happening. :thinking:

**Comment by wangyang0918** [comment]
I could reproduce this issue by running about more than 500 times locally. Maybe the `FutureUtils.retry` run too fast in your machine :)

**Comment by wangyang0918** [comment]
@flinkbot run azure

**Comment by wangyang0918** [comment]
@flinkbot run azure

**Comment by wangyang0918** [comment]
@flinkbot run azure

**Comment by wangyang0918** [comment]
All the failed pipelines are caused by FLINK-29121.

It is unrelated to this PR. Merging now.

**Pull request merged**